swift/Tests/Blockchains/CosmosTests.swift

```diff
@@ -151,4 +151,53 @@ class CosmosSignerTests: XCTestCase {
         XCTAssertEqual(output.serialized, "CukDCqABCjcvY29zbW9zLmRpc3RyaWJ1dGlvbi52MWJldGExLk1zZ1dpdGhkcmF3RGVsZWdhdG9yUmV3YXJkEmUKLWNvc21vczEwMHJoeGNscWFzeTZ2bnJjZXJ2Z2g5OWFseDV4dzdsa2ZwNHU1NBI0Y29zbW9zdmFsb3BlcjFleTY5cjM3Z2Z4dnhnNjJzaDRyMGt0cHVjNDZwempybTg3M2FlOAqgAQo3L2Nvc21vcy5kaXN0cmlidXRpb24udjFiZXRhMS5Nc2dXaXRoZHJhd0RlbGVnYXRvclJld2FyZBJlCi1jb3Ntb3MxMDByaHhjbHFhc3k2dm5yY2VydmdoOTlhbHg1eHc3bGtmcDR1NTQSNGNvc21vc3ZhbG9wZXIxc2psbHNucmFtdGczZXd4cXd3cndqeGZnYzRuNGVmOXUybGNuajAKoAEKNy9jb3Ntb3MuZGlzdHJpYnV0aW9uLnYxYmV0YTEuTXNnV2l0aGRyYXdEZWxlZ2F0b3JSZXdhcmQSZQotY29zbW9zMTAwcmh4Y2xxYXN5NnZucmNlcnZnaDk5YWx4NXh3N2xrZnA0dTU0EjRjb3Ntb3N2YWxvcGVyMTY0OHlubHBkdzdmcWEyYXh0MHcyeXAzZms1NDJqdW5sN3JzdnE2EmUKUQpGCh8vY29zbW9zLmNyeXB0by5zZWNwMjU2azEuUHViS2V5EiMKIQJXKG7D830zVXu7qgALJ3RKyQI6qZZ8rnWhgdH/kfqdxRIECgIIARi+AhIQCgoKBXVhdG9tEgExEOC2DRpAXLgJ+8xEMUn7nkFj3ukg2V65Vh5ob7HKeCaNpMM6OPQrpW2r6askfssIFcOd8ThiBEz65bJz81Fmb5MtDTGv4g==")
         XCTAssertEqual(output.error, "")
     }
+
+    func testIbcTransfer() {
+        let privateKey = PrivateKey(data: Data(hexString: "8bbec3772ddb4df68f3186440380c301af116d1422001c1877d6f5e4dba8c8af")!)!
+        let publicKey = privateKey.getPublicKeySecp256k1(compressed: true)
+        let fromAddress = AnyAddress(publicKey: publicKey, coin: .cosmos)
+
+        let transferMessage = CosmosMessage.Transfer.with {
+            $0.sourcePort = "transfer"
+            $0.sourceChannel = "channel-141"
+            $0.sender = fromAddress.description
+            $0.receiver = "osmo18s0hdnsllgcclweu9aymw4ngktr2k0rkvn7jmn"
+            $0.token = CosmosAmount.with {
+                $0.amount = 100000
+                $0.denom = "uatom"
+            }
+            $0.timeoutHeightRevisionNumber = 1
+            $0.timeoutHeightRevisionHeight = 8800000
+            $0.timeoutTimestamp = 0
```

Comment:
those number looks magic to me, maybe we can have some default values?

Reply:
Unfortunately block height cannot be constant, so it must come from client.  I've added a separate note on this, and also extended proto code comments.